packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -1759,19 +1759,20 @@ export class Domain extends DomainBase implements IDomain, ec2.IConnectable {
  *
  * Domain endpoints look like this:
  *
- *   https://example-domain-jcjotrt6f7otem4sqcwbch3c4u.us-east-1.es.amazonaws.com
- *   https://<domain-name>-<suffix>.<region>.es.amazonaws.com
+ *   https://search-example-domain-jcjotrt6f7otem4sqcwbch3c4u.us-east-1.es.amazonaws.com
+ *   https://search-<domain-name>-<suffix>.<region>.es.amazonaws.com
  *
  * ..which means that in order to extract the domain name from the endpoint, we can
- * split the endpoint using "-<suffix>" and select the component in index 0.
+ * split the endpoint using "-<suffix>" and select the component in index 0, and then replace
+ * the "search-" prefix.
  *
  * @param domainEndpoint The Elasticsearch domain endpoint
  */
 function extractNameFromEndpoint(domainEndpoint: string) {
   const { hostname } = new URL(domainEndpoint);
   const domain = hostname.split('.')[0];
   const suffix = '-' + domain.split('-').slice(-1)[0];
-  return domain.split(suffix)[0];
+  return domain.split(suffix)[0].replace('search-', '');
```

Comment:
Can we also consider ES Domains with VPC endpoints? In this case, the prefix is `vpc-` instead of `search-`.

Reply:
Suggested approach to avoid clashing with the actual domain name:

```
function extractNameFromEndpoint(domainEndpoint: string) {
  const { hostname } = new URL(domainEndpoint);
  const domain = hostname.split('.')[0];
  const suffix = '-' + domain.split('-').slice(-1)[0];
  const prefix = domain.split('-')[0] + '-';
  return domain.slice(prefix.length).split(suffix)[0];
}
```